datafusion/expr/src/expr.rs

```diff
@@ -1574,8 +1574,13 @@ impl Expr {
 
     /// Returns true if the expression is volatile, i.e. whether it can return different
     /// results when evaluated multiple times with the same input.
-    pub fn is_volatile(&self) -> Result<bool> {
-        self.exists(|expr| Ok(expr.is_volatile_node()))
+    ///
+    /// For example the function call `RANDOM()` is volatile as each call will
+    /// return a different value.
+    ///
+    /// See [`Volatility`] for more information.
+    pub fn is_volatile(&self) -> bool {
+        self.exists(|expr| Ok(expr.is_volatile_node())).unwrap()
```

Comment:
Add code comment why unwrap is safe here.

btw maybe we can have `exists` variant which doesn't throw?

Reply:
I will add the comment in a follow on PR The non fallable exists variant is an interesting idea -- maybe we can file a follow on issue for it

Reply:
Follow on PR: https://github.com/apache/datafusion/pull/13217